sdk/storage/azure-storage-blob/azure/storage/blob/_list_blobs_helper.py

```diff
@@ -226,74 +222,72 @@ class BlobPrefix(ItemPaged, DictMixin):
-    :ivar int results_per_page: The maximum number of results retrieved per API call.
-    :ivar str next_marker: The continuation token to retrieve the next page of results.
-    :ivar str location_mode: The location mode being used to list results. The available
-        options include "primary" and "secondary".
-    :ivar current_page: The current page of listed results.
-    :vartype current_page: list(~azure.storage.blob.BlobProperties)
-    :ivar str container: The container that the blobs are listed from.
-    :ivar str delimiter: A delimiting character used for hierarchy listing.
-
-    :param callable command: Function to retrieve the next page of items.
-    :param str prefix: Filters the results to return only blobs whose names
-        begin with the specified prefix.
-    :param int results_per_page: The maximum number of blobs to retrieve per
-        call.
-    :param str marker: An opaque continuation token.
-    :param str delimiter:
-        Used to capture blobs whose names begin with the same substring up to
-        the appearance of the delimiter character. The delimiter may be a single
-        character or a string.
-    :param location_mode: Specifies the location the request should be sent to.
-        This mode only applies for RA-GRS accounts which allow secondary read access.
-        Options include 'primary' or 'secondary'.
-    """
-    def __init__(self, *args, **kwargs):
+    Can be thought of as a virtual blob directory."""
+
+    name: Optional[str]
+    """The prefix, or "directory name" of the blob."""
+    service_endpoint: Optional[str]
+    """The service URL."""
+    prefix: Optional[str]
+    """A blob name prefix being used to filter the list."""
+    marker: Optional[str]
+    """The continuation token of the current page of results."""
+    results_per_page: Optional[int]
+    """The maximum number of results retrieved per API call."""
+    next_marker: Optional[str]
+    """The continuation token to retrieve the next page of results."""
+    location_mode: Optional[str]
+    """The location mode being used to list results. The available
+    options include "primary" and "secondary"."""
+    current_page: Optional[List[BlobProperties]]
+    """The current page of listed results."""
+    delimiter: str
+    """A delimiting character used for hierarchy listing."""
+    command: Callable
+    """Function to retrieve the next page of items."""
+    container: Optional[str]
+    """The name of the container."""
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
         super(BlobPrefix, self).__init__(*args, page_iterator_class=BlobPrefixPaged, **kwargs)
         self.name = kwargs.get('prefix')
         self.prefix = kwargs.get('prefix')
         self.results_per_page = kwargs.get('results_per_page')
         self.container = kwargs.get('container')
-        self.delimiter = kwargs.get('delimiter')
+        self.delimiter = kwargs.get('delimiter', '/')
```

Comment:
This seems like a sketchy change. We should either mark delimiter as `Optional` or add an ignore here, depending on if delimiter can ever be `None` when returned.

Reply:
Hmm yeah so I think my logic here is that a valid `BlobPrefix` is only when a delimiter is provided, and in any instances where we call `walk_blobs`, `/` is provided as a default delimiter. So I think my idea here is that it should be non-None, and if it is None then on the `walk_blobs` side it should have been defaulted to `/`. _BUT_ this logic is wrong in putting it as part of the kwarg pop because that means they specifically passed in `None` (and so while that may cause an exception, I shoudn't be swallowing it and transforming it into a `/` and trying anyways)

tl;dr I had bad logic thinking about this change but now realize what I should do 😄